.devcontainer/.pre-commit-config.yaml

```diff
@@ -59,6 +57,18 @@ repos:
         types: [shell]
         args: ["-e", "SC1091"] # Optional: exclude specific checks
         # args: ["--severity=warning"]  # Optional: show only errors and warnings
+      - id: pytest
+        name: Run Pytest Tests
+        entry: poetry run pytest --dist=loadfile -n $(nproc)
+        language: system
+        types: [python]
+        description: Run pytest tests in parallel using xdist
```

Comment:
@amha-kindu it's seeems that @coderabbitai is hallucinating here buy overall having a run duration limit might be a good idea. It also might worth to look at `distributed and subprocess testing` section of help and make necessary changes so it's able to scare itself properly on default setting:

```
distributed and subprocess testing:
  -n numprocesses, --numprocesses=numprocesses
                        Shortcut for '--dist=load --tx=NUM*popen'.
                        With 'logical', attempt to detect logical CPU count (requires psutil, falls back to
                        'auto').
                        With 'auto', attempt to detect physical CPU count. If physical CPU count cannot be
                        determined, falls back to 1.
                        Forced to 0 (disabled) when used with --pdb.
...
```

Reply:
> @amha-kindu it's seeems that @coderabbitai is hallucinating here buy overall having a run duration limit might be a good idea. It also might worth to look at `distributed and subprocess testing` section of help and make necessary changes so it's able to scare itself properly on default setting:
> 
> ```
> distributed and subprocess testing:
>   -n numprocesses, --numprocesses=numprocesses
>                         Shortcut for '--dist=load --tx=NUM*popen'.
>                         With 'logical', attempt to detect logical CPU count (requires psutil, falls back to
>                         'auto').
>                         With 'auto', attempt to detect physical CPU count. If physical CPU count cannot be
>                         determined, falls back to 1.
>                         Forced to 0 (disabled) when used with --pdb.
> ...
> ```

Thanks for pointing this out @alexanderilyin. I'll introduce about 300 sec time limit. Regarding the distributed settings, I've checked pytest-xdist documentation as well as pytest help on the terminal and I'll use `-n auto` so that it scales itself properly on default settings